### ISSUE TITLE
[BUG] Unable to communicate with gate (pyluos 2.2.9)

### ISSUE THREAD

**Issue by SonDePoisson** [pollen-robotics/pyluos#178]
**Describe the bug**
Hello, 
With a gate exemple pyluos-shell or bootloader never find the gate with pyluos 2.2.9. I have not this problem with pyluos 2.2.8 and previous versions.

**Network configuration**
Using a Nucleo G431KB in gate with Luos engine 2.2.8 and powered by usb.

**How to reproduce the bug**
Upload a gate exemple and try to execute pyluos-shell

**Additional context**
```
~/Documents/Luos_Tuto/luos_engine/examples/projects/NUCLEO-G431KB/gate_serialcom (main) » pyluos-bootloader detect /dev/luos-gate > ~/luos.log  130 ↵ toto@toto-gs
Fatal Python error: Cannot recover from stack overflow.
Python runtime state: initialized

Thread 0x00007fceb8ea4700 (most recent call first):
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/serial_io.py", line 125 in _poll
  File "/usr/lib/python3.8/threading.py", line 870 in run
  File "/usr/lib/python3.8/threading.py", line 932 in _bootstrap_inner
  File "/usr/lib/python3.8/threading.py", line 890 in _bootstrap

Current thread 0x00007fcebbfd8740 (most recent call first):
  File "/usr/lib/python3.8/threading.py", line 264 in _is_owned
  File "/usr/lib/python3.8/threading.py", line 293 in wait
  File "/usr/lib/python3.8/queue.py", line 179 in get
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/serial_io.py", line 68 in recv
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 18 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
  File "/home/toto/.local/lib/python3.8/site-packages/pyluos/io/__init__.py", line 24 in read
```

**Comment by SonDePoisson** [comment]
To fix this problem : 
add the following line back after line 18 in _pyluos/io/\_\_init\_\_.py_ 
```python
return self.loads(data)
```

**Comment by nicolas-rabault** [comment]
Hi @SonDePoisson thank you for this.
Would you be interested by fixing this yourself by Pull Request to contribute to Luos?

**Comment by JeromeGalan** [comment]
Thank you @SonDePoisson.
It's fixed in Pyluos version 2.2.10
